packages/ui/src/components/vuestic-mixins/FormComponent/cFormComponent.ts

```diff
@@ -0,0 +1,174 @@
+import { ref, toRefs, computed, inject, onMounted, onUnmounted } from 'vue'
+import { FormProvider, FormServiceKey, FormElement } from '../../vuestic-components/va-form/consts'
+import isString from 'lodash/isString'
+import isFunction from 'lodash/isFunction'
+import flatten from 'lodash/flatten'
+
+const prepareValidations = (messages: string | any[] = [], callArguments = null) => {
+  if (isString(messages)) {
+    return [messages]
+  }
+  return messages
+    .map((message) => isFunction(message) ? message(callArguments) : message)
+}
+
+export function useFormProvider (context: FormElement) {
+  const formProvider: FormProvider | undefined = inject(FormServiceKey, undefined)
+  onMounted(() => {
+    if (formProvider?.onChildMounted) {
+      formProvider.onChildMounted(context)
+    }
+  })
+  onUnmounted(() => {
+    if (formProvider?.onChildUnmounted) {
+      formProvider.onChildUnmounted(context)
+    }
+  })
+}
+
+export function useFormComponent (props: Record<any, any>, focus: () => void, reset: () => void) {
+  const hadFocus = ref(false)
+  const isFocused = ref(false)
+  const internalErrorMessages = ref([] as Array<any>)
+  const internalError = ref(false)
+  const isFormComponent = ref(true)
+  const { error, errorMessages, rules, modelValue } = toRefs(props)
+
+  useFormProvider({
+    resetValidation,
+    hasError,
+    validate,
+    focus,
+    reset,
+  })
+
+  const shouldValidateOnBlur = computed<boolean>(() => hadFocus.value)
+
+  const computedError = computed<boolean>({
+    get () {
+      return error.value || internalError.value
+    },
+    set (value) {
+      internalError.value = value
+    },
+  })
+
+  const computedErrorMessages = computed<Array<any>>({
+    get () {
+      return errorMessages.value ? prepareValidations(errorMessages) : internalErrorMessages.value
+    },
+    set (errorMessages) {
+      internalErrorMessages.value = errorMessages
+    },
+  })
+
+  function validate () {
+    computedError.value = false
+    computedErrorMessages.value = []
+
+    if (rules.value && rules.value.length > 0) {
+      prepareValidations(flatten(rules), modelValue.value)
+        .forEach((validateResult) => {
+          if (isString(validateResult)) {
+            computedErrorMessages.value.push(validateResult)
+            computedError.value = true
+          } else if (validateResult === false) {
+            computedError.value = true
+          }
+        })
+    }
+
+    return !computedError.value
+  }
+
+  function resetValidation (): void {
+    computedErrorMessages.value = []
+    computedError.value = false
+  }
+
+  function hasError (): boolean {
+    return computedError.value
+  }
+
+  // eslint-disable-next-line camelcase
+  function ValidateMixin_onBlur (): void {
```

Comment:
It is not ValidationMixin in `useFormComponent` I think. You can simply rename it to `onBlur`.

Reply:
Also, you can create formComponentListeners = { onBlur, onFocus }, then we can use it as v-bind="formComponentListeners".